Runner/CustomTemplateFactory.py

```diff
@@ -0,0 +1,170 @@
+import json
+"""
+This module responsiblilty is for reading and setting elements of the json templates
+"""
+
+
+def set_template_name(template, pool_id):
+    try:
+        template["parameters"]["poolName"]["defaultValue"] = pool_id
+    except KeyError:
+        pass
+    try:
+        template["parameters"]["poolId"]["defaultValue"] = pool_id
+    except KeyError:
+        pass
+    try:
+        template["parameters"]["poolId"]["value"] = pool_id
+    except KeyError:
+        pass
+
+
+def set_parameter_name(template, job_id):
+    try:
+        template["jobName"]["value"] = job_id
+    except KeyError:
+        pass
+    try:
+        template["jobId"]["value"] = job_id
+    except KeyError:
+        pass
+
+
+def set_parameter_storage_info(template, storage_info):
+    # Set input filegroup
+    """
+    'fgrp-' needs to be removed.
+    """
+    try:
+        template["inputData"]["value"] = storage_info.input_container.replace("fgrp-", "")
+    except KeyError:
+        pass
+    try:
+        template["inputFilegroup"]["value"] = storage_info.input_container.replace("fgrp-", "")
+    except KeyError:
+        pass
+
+    # Set file group SAS input
+    try:
+        template["inputFilegroupSas"]["value"] = storage_info.input_container_SAS
+    except KeyError:
+        pass
+    try:
+        template["inputDataSas"]["value"] = storage_info.input_container_SAS
+    except KeyError:
+        pass
+
+    # Set output filegroup
+    try:
+        template["outputFilegroup"]["value"] = storage_info.output_container.replace("fgrp-", "")
```

Comment:
could the container have `frgp-` somewhere else in the name. If so maybe using a regex might be better